### DISCUSSION TITLE
Unable to compile TypeScript with SocketIO

### DISCUSSION THREAD

**Discussion by Tungtono** [feathersjs/feathers#3131] [Q&A]
### Steps to reproduce

I'm trying out Feathers 4.5.15, app generated from feathers CLI with Typescript. Just added `@feathersjs/socketio`, also 4.5.15 and configured with `app.configure(socketio());`, but unable to compile with the type error:
```
[ERROR] 16:15:18 ⨯ Unable to compile TypeScript:
src/app.ts(49,15): error TS2345: Argument of type '(app: Application<any, any>) => void' is not assignable to parameter of type '(this: Application, app: Application) => void'.
  Types of parameters 'app' and 'app' are incompatible.
    Type 'Application' is not assignable to type 'Application<any, any>'.
```
am I using the right version of `@feathersjs/socketio` or what should I do with this??

### Expected behavior

Tell us what should happen

### Actual behavior

Tell us what happens instead

### System configuration

Tell us about the applicable parts of your setup.

**Module versions** (especially the part that's not working): 4.5.15

**NodeJS version**: 16.0.0

**Operating System**: 

**Browser Version**:

**React Native Version**:

**Module Loader**:

**Comment by daffl** [comment]
You probably have a version mismatch somewhere. New apps should be generated with the latest version which guarantees that everything will work as intended.

**Reply by Tungtono** [reply]
That app I initiated a couple of months ago, I guess before the latest version 5 was official so it used 4.5.15. I set up a few things and left it there, only until today I installed `feathers-socketio`, tried both the latest ver 5. and back to 4.5.15 but it didn't work. Ended up having to go with a `/* @ts-ignore */` for now. 

Do you happen to recall which version of `feathers-socketio` was compatible with feathersjs 4.5.15?

**Reply by daffl** [reply]
The generator itself should still use the correct versions. What does the dependency list in your package.json look like?

**Reply by Tungtono** [reply]
yeah, here's the list

```
{
    "@feathersjs/authentication": "^4.5.15",
    "@feathersjs/authentication-local": "^4.5.15",
    "@feathersjs/authentication-oauth": "^4.5.15",
    "@feathersjs/configuration": "^4.5.15",
    "@feathersjs/errors": "^4.5.15",
    "@feathersjs/express": "^4.5.15",
    "@feathersjs/feathers": "^4.5.15",
    "@feathersjs/socketio": "^4.5.15",
    "@feathersjs/transport-commons": "^4.5.15",
    "@types/compression": "^1.7.2",
    "@types/cors": "^2.8.13",
    "@types/jest": "^29.2.4",
    "@types/jsonwebtoken": "^9.0.0",
    "@types/nodemailer": "^6.4.7",
    "@types/serve-favicon": "^2.5.3",
    "@types/validator": "^13.7.10",
    "@typescript-eslint/eslint-plugin": "^5.47.1",
    "@typescript-eslint/parser": "^5.47.1",
    "axios": "^1.2.2",
    "compression": "^1.7.4",
    "cors": "^2.8.5",
    "eslint": "^8.30.0",
    "feathers-authentication-management": "^4.0.2",
    "feathers-hooks-common": "^6.1.5",
    "feathers-sequelize": "^6.4.0",
    "helmet": "^5.1.1",
    "nodemailer": "^6.9.0",
    "pg": "^8.8.0",
    "sequelize": "^6.28.0",
    "serve-favicon": "^2.5.0",
    "shx": "^0.3.4",
    "ts-jest": "^29.0.3",
    "ts-node-dev": "^2.0.0",
    "typescript": "^4.9.4",
    "winston": "^3.8.2"
    "@types/bluebird": "^3.5.38",
    "jest": "^29.3.1"
}
```
